internal/pkg/agent/install/install.go

```diff
@@ -75,12 +75,29 @@ func Install(cfgFile, topPath string, unprivileged bool, log *logp.Logger, pt *p
 
 	pt.Describe("Copying install files")
 	copyConcurrency := calculateCopyConcurrency(streams)
-	err = copyFiles(copyConcurrency, pathMappings, dir, topPath)
+
+	skipFn := func(relPath string) bool { return false }
+	if flavor != "" {
+		flavorDefinition, err := Flavor(flavor, "", manifest.Package.Flavors)
+		if err != nil {
+			return utils.FileOwner{}, err
+		}
+		skipFn, err = SkipComponentsPathFn(paths.VersionedHome(dir), flavorDefinition)
+		if err != nil {
+			return utils.FileOwner{}, err
+		}
+	}
+
+	err = copyFiles(copyConcurrency, pathMappings, dir, topPath, skipFn)
 	if err != nil {
 		pt.Describe("Error copying files")
 		return utils.FileOwner{}, err
 	}
 
+	if err := markFlavor(topPath, flavor); err != nil {
+		return utils.FileOwner{}, err
```

Comment:
nit: wrap the error with some context, same goes for the other new locations we can return errors from the install command.